pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -194,8 +205,8 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 		return nil
 	}
 
-	if err := ValidateTaskRunAndTask(tr.Spec.Inputs.Params, rtr); err != nil {
-		c.Logger.Error("Failed to validate taskrun %s with error %v", tr.Name, err)
+	if err := ValidateResolvedTaskResources(tr.Spec.Inputs.Params, rtr); err != nil {
```

Comment:
can `validateResolvedTaskResources` be part of `ResolveTaskResources`? we don't want to get back invalid resources

Reply:
I really liked keeping these separate tbh! The logic required to follow all the references and find the objects I think is pretty different from the logic required to make sure all the required resources + params are present.

And we can have two different errors, one where you referred to something that couldn't be retrieved, and one where you didn't provide something that was needed

Reply:
fair point about different errors, we can always optimize later if we find out we need to